waku/waku_filter_v2/protocol.nim

```diff
@@ -295,6 +307,7 @@ proc new*(
       subscriptionTimeout, maxFilterPeers, maxFilterCriteriaPerPeer
     ),
     peerManager: peerManager,
+    messageCache: init(TimedCache[string], timeout = 2.minutes),
```

Comment:
I would personally add the cache timeout as an input argument to the WakuFilter's new because it enables configurability even if most cases default is choosern.
Also it enables to create very easy unit tests that checks the feature correctness automatically.
So in unit test this way you can apply 1.seconds timeout which is more than enough to check such dup cases.
WDYT?

Reply:
Yes, this feels like a hard-coded solution. Let's set 2 minutes as the default and allow the Waku filter to modify it as needed.